### PR TITLE
Change project submission form for 2017

### LINKED ISSUE #551: Meta: project proposal 2017
* [x] project submission form: "issues and features" will be renamed to "what will the students be working on?" to make it more general
* [x] project submission form: a new text area will be added with the title "requirements" (or something similar) that will list requirements for the project from the point of view of the mentor (e.g. good knowledge of a language, proficiency with CSS, etc..). This can include a little coding challenge or a codecademy track that needs to be completed; more examples of this will be provided in the mentor guide.
* [X] project submission form: add link to guide at the top of the form, to make sure they've read it

### PR DESCRIPTION
**Pull request by carpodaster**
This closes #551 

I would especially appreciate some feedback on the wording of the help text (placeholder values). I've also deployed it to [staging](https://rgsoc-teams-staging.herokuapp.com) so you can play around with it.

### CODE REVIEW

File: app/views/projects/new.html.slim [hunk 1]
```diff
@@ -31,11 +31,12 @@ p.well Please make sure you have read our #{link_to "mentor guide", 'http://rail
       = f.input :source_url, as: :url, wrapper_html: { class: 'col-md-6' }, label: "Repository", placeholder: 'Link to your source code (Github, Gitlab, Bitbucket, …)' 
       = f.input :description, wrapper_html: { class: 'col-md-6' }, placeholder: "What's your project about?", input_html: { rows: 5 }, hint: 'You can use basic <a href="http://en.wikipedia.org/wiki/Markdown" target="_blank">Markdown</a> here.'.html_safe
       = f.input :issues_and_features, wrapper_html: { class: 'col-md-6' }, placeholder: "A list of possible features to work on, maybe even concrete Github issues", input_html: { rows: 5 }, hint: 'You can use basic <a href="http://en.wikipedia.org/wiki/Markdown" target="_blank">Markdown</a> here.'.html_safe
+      = f.input :requirements, wrapper_html: { class: 'col-md-6' }, placeholder: "E.g. good knowledge of a specific programming language, proficiency with CSS, etc. Maybe a coding challenge (on Exercism.io, Codeschool, Codewars, etc.) that needs to be completed.", input_html: { rows: 6 }, hint: 'You can use basic <a href="http://en.wikipedia.org/wiki/Markdown" target="_blank">Markdown</a> here.'.html_safe
```
Comment (F3PiX) [inline app/views/projects/new.html.slim]: "You can list requirements here, e.g. skill level of a programming language or framework. You can illustrate the required skill level with a relevant coding challenge that needs to be completed (on Exercism.io, Codeschool course, Codewars or something of your own choice.) "

**Comment by F3PiX** [comment]
About the link to the mentor guide: how about adding an [acceptance validation](http://guides.rubyonrails.org/active_record_validations.html#acceptance)? So they need to check a checkbox to confirm that they read the guide.

File: app/views/projects/new.html.slim [hunk 2]
```diff
@@ -31,11 +31,12 @@ p.well Please make sure you have read our #{link_to "mentor guide", 'http://rail
       = f.input :source_url, as: :url, wrapper_html: { class: 'col-md-6' }, label: "Repository", placeholder: 'Link to your source code (Github, Gitlab, Bitbucket, …)' 
       = f.input :description, wrapper_html: { class: 'col-md-6' }, placeholder: "What's your project about?", input_html: { rows: 5 }, hint: 'You can use basic <a href="http://en.wikipedia.org/wiki/Markdown" target="_blank">Markdown</a> here.'.html_safe
       = f.input :issues_and_features, wrapper_html: { class: 'col-md-6' }, placeholder: "A list of possible features to work on, maybe even concrete Github issues", input_html: { rows: 5 }, hint: 'You can use basic <a href="http://en.wikipedia.org/wiki/Markdown" target="_blank">Markdown</a> here.'.html_safe
```
Comment (klappradla) [inline app/views/projects/new.html.slim]: I suggest to wide the scope of this question - by making the hint more general 😉 
How about:
> How will the students be involved? What will they be working on? Possible features, maybe links to concrete issues on Github.

I liked how maintainers came up with special *"projects"* within their projects for RGSoC including guidance, etc. that's why I think *"How will the students be involved"* could also help here to a) get more meaningful answers and b) have the mentors think about this in advance.

File: app/views/projects/new.html.slim (hunk 1 above)
Comment (klappradla) [inline app/views/projects/new.html.slim]: I'm for some reason pro avoiding *"you"* in hints 🙈  - but I really like your point about
> something of your own choice

and
> skill level of programming language

However, I don't think it's necessary to repeat *"you can list requirements"* as it's already the title of the field.

How about something like:
> Skills students should ideally have - e.g. good knowledge of a specific programming language, proficiency with CSS, etc. Maybe illustrate via a coding challenge that needs to be completed (on Exercism.io, Codeschool, Codewars or somewhere else).

I deliberately would not put *"framework"* there, as it gets especially annoying in the JS world where they drop another one every 24hours and I feel forcing people to look at an new DSL for the same concepts all the time just puts additional pressure. Feel free to say I'm totally wrong.

Comment (F3PiX) [inline app/views/projects/new.html.slim]: :-) I like your text. I am not sure about 'ideally' - I feel it is about the minimal requirements. So, maybe it is a good idea to name the two things: basic requirements, and ideal.

**Comment by klappradla** [comment]
Added a tiny change to the besides that awesome *"hint"* (or however you call elements like this) above the form: I think it could be of use to already be consistent about the rebranding goals here. Especially around the project submission form, the word **"Rails"** could be kinda distracting for mentors 😉

**Comment by klappradla** [comment]
Hm, I personally don't really see the discussion here as "done" - at least I would have loved to get @carpodaster 's feedback before... Why did you merge already @nbogis ?

**Comment by F3PiX** [comment]
Test to see if we can continue the conversation here.

**Comment by nbogis** [comment]
sorry I did it by mistake. I thought that button merges the changes with my stuff not closing the PR. My bad

**Pull request merged**